File: split/terminput.pas
procedure terminput;
var
  k: 0..bufsize;
begin
  break(termout);
  if not inputln(termin, true) then
    fatalerror(261);
  termoffset := 0;
  selector := selector - 1;
  if last <> First then
    for k := First to last - 1 do
      print(buffer[k]);
  println;
  selector := selector + 1;
end;

File: cpp/src/terminput.cpp
#include "terminput.h"
#include "inputln.h"
#include "impression.h"
#include "fatalerror.h"
#include <iostream>

void terminput(void)
{
	std::cout << std::flush;
	if (!inputln(std::cin, true))
		fatalerror("End of file on the terminal!");
	termoffset = 0;
	selector--;
	print(std::string(buffer+First, buffer+last)+"\n");
	selector++;
}
